### PR TITLE
US129595 - Add Low Mean Card to Summary Section of QIA Page

### PR DESCRIPTION
**Pull request by JikaiLong**
[US129595](https://rally1.rallydev.com/#/?detail=/userstory/602885916475&fdp=true): [AQ] Add Low Mean Card to Summary Section of QIA Page

* Functional Testing 
  * [x] Browsers (Chrome, FF, Edge)
  * [x] Accessibility
  * [x] RTL
  * [x] Responsive
  * [x] Unit tests
* Security, devops, perf: N/A
* UX and docs
  * [x] Demo to Kevin

Also updated quiz detail card to match result card size. 
URL state will be updated in a separated story. 
  
FYI @Brightspace/draco-vis

### Screenshot
image

(skeleton)
image

### CODE REVIEW

**Comment by JikaiLong** [comment]
Quiz Data is not quite working, fixing right now.

File: model/filteredData.js [hunk 1]
```diff
@@ -48,6 +48,11 @@ export class FilteredData {
 		return this._data.quizzes.filter(r => appliedFilters.every(f => f.filter(r, this.userDictionary)));
 	}
 
+	get questions() {
+		const appliedFilters = this.filters.filter(f => f.isApplied);
+		return this._data.quizItemData.questions.filter(r => appliedFilters.every(f => f.filter(r, this.userDictionary)));
```
Comment (Vitalii-Misechko) [inline model/filteredData.js:53]: Could you delete `this.userDictionary` here and at line 48? It is not defined in AQ it's just copy-paste from EQ?

File: components/cards/low-mean-card.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	static get properties() {
+		return {
+			data: { type: Object, attribute: false },
+			wide: { type: Boolean, attribute: true },
+			tall: { type: Boolean, attribute: true },
+			lowMeanThreshold: { type: Number, attribute: 'low-mean-threshold' }
+		};
+	}
+
+	constructor(lowMeanThreshold = 0.4) {
```
Comment (Vitalii-Misechko) [inline components/cards/low-mean-card.js:46]: There is no code that passes a value to this constructor.  Could you delete the parameter then?

File: pages/quiz-item-analysis-view.js [hunk 3]
```diff
@@ -466,6 +496,26 @@ class QuizItemAnalysisView extends SkeletonMixin(Localizer(MobxLitElement)) {
...
+
+	_lowMeanCard({ wide, tall, skeleton }) {
+		return html`
+			<d2l-insights-low-mean-card
+				.data = ${this._filteredData} 
+				id="d2l-insights-assessment-low-mean"
+				?skeleton="${skeleton}"
+				?wide="${wide}" 
+				?tall="${tall}"
+			></d2l-insights-low-mean-card>
```
Comment (Vitalii-Misechko) [inline pages/quiz-item-analysis-view.js:515]: You should pass `low-mean-threshold` value here and all the way up to the demo page

File: test/components/cards/low-mean-card.test.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+			expect(el.shadowRoot.querySelector('d2l-labs-summary-card').message).to.equal('Quiz items with a mean of 40 % or lower');
+		});
+
+		it('should render the proper value for empty quizzes', async() => {
+			const el = await fixture(html`<d2l-insights-low-mean-card .data="${emptyQuestion}"></d2l-insights-low-mean-card>`);
+			expect(el.shadowRoot.querySelector('d2l-labs-summary-card').value).to.equal('0');
+			expect(el.shadowRoot.querySelector('d2l-labs-summary-card').title).to.equal('Low Mean');
+			expect(el.shadowRoot.querySelector('d2l-labs-summary-card').message).to.equal('Quiz item with a mean of 40 % or lower');
+		});
+	});
```
Comment (Vitalii-Misechko) [inline test/components/cards/low-mean-card.test.js:52]: nice tests!

File: components/cards/low-mean-card.js (hunk 2 above)
Comment (JikaiLong) [reply components/cards/low-mean-card.js:46]: I think I want to leave it here for a better setting page integration,

File: pages/quiz-item-analysis-view.js (hunk 3 above)
Comment (JikaiLong) [reply pages/quiz-item-analysis-view.js:515]: I will leave it like this for now like other cards I think.

File: components/cards/low-mean-card.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+import { formatPercent } from '@brightspace-ui/intl';
+import { html } from 'lit-element';
+import { Localizer } from '../../locales/localizer';
+import { MobxLitElement } from '@adobe/lit-mobx';
+import { SkeletonMixin } from '@brightspace-ui/core/components/skeleton/skeleton-mixin.js';
+import { ToastEvent } from '@brightspace-ui-labs/toaster';
+
+export const LOW_MEAN_FILTER_ID = 'd2l-insights-low-mean-card';
+
+export class LowMeanFilter {
```
[the remark this answers is not in the document]
Comment (JikaiLong) [reply components/cards/low-mean-card.js:11]: gonna update both low mean card and average quiz score card

File: components/cards/low-mean-card.js [hunk 6]
```diff
@@ -9,9 +9,9 @@ import { ToastEvent } from '@brightspace-ui-labs/toaster';
 export const LOW_MEAN_FILTER_ID = 'd2l-insights-low-mean-card';
 
 export class LowMeanFilter {
-	constructor(lowMeanThreshold = 0.4) {
+	constructor() {
 		this.isApplied = false;
-		this.lowMeanThreshold = lowMeanThreshold;
+		this.lowMeanThreshold = 0.4;
```
Comment (Vitalii-Misechko) [inline components/cards/low-mean-card.js:14]: `nit` could you use the value from `settingsDefaults`?

File: components/cards/low-mean-card.js [hunk 7]
```diff
@@ -43,12 +43,12 @@ class LowMeanCard extends SkeletonMixin(Localizer(MobxLitElement)) {
...
 	}
 
-	constructor(lowMeanThreshold = 0.4) {
+	constructor() {
 		super();
 		this.data = {};
 		this.wide = false;
 		this.tall = false;
-		this.lowMeanThreshold = lowMeanThreshold;
+		this.lowMeanThreshold = 0.4;
```
Comment (Vitalii-Misechko) [inline components/cards/low-mean-card.js:51]: `nit` could you use the value from `settingsDefaults`?

**Review by Vitalii-Misechko — APPROVED** [review]
LGTM! Thank you for improvements

**Review by NicholasHallman — APPROVED** [review]
LGTM :)

**Comment by JikaiLong** [comment]
I like how the url setting is persisted when you going back and forth between the main page and QIA page. Awesome. @NicholasHallman

**Pull request merged**